app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,93 @@
+class PaymentsController < ApplicationController
+  require 'hips'
+
+  protect_from_forgery except: :webhook
+
+  def create
+    payment_type = params[:type]
+    user_id = params[:user_id]
+
+    @payment = Payment.create(payment_type: payment_type,
+                              user_id: user_id,
+                              status: Payment.order_to_payment_status(nil))
+
+    urls = { success: payment_success_url(user_id: user_id, id: @payment.id) }
+    urls[:error]   = payment_error_url(user_id: user_id, id: @payment.id)
+
+    urls[:webhook] = (SHF_WEBHOOK_HOST || root_url) +
+                     payment_webhook_path.sub('/en', '')
+
+    payment_data = { id: @payment.id, type: payment_type, currency: 'SEK' }
+
+    hips_order = HipsService.create_order(user_id,
+                                          session.id,
+                                          payment_data,
+                                          urls)
+    @hips_id = hips_order['id']
+    @payment.hips_id = @hips_id
+    @payment.status = Payment.order_to_payment_status(hips_order['status'])
+    @payment.save!
+
+  rescue RuntimeError, HTTParty::Error, ActiveRecord::RecordInvalid  => exc
+    @payment.destroy if @payment.persisted?
+
+    log_hips_activity('create order', 'error', nil, @hips_id, exc)
+
+    log_hips_activity('create order', 'error', nil, @hips_id, exc.cause)
+
+    helpers.flash_message(:alert, t('.something_wrong'))
+
+    redirect_back fallback_location: root_path
+  end
+
+  def webhook
+    # This webhook will be called multiple times (7) during the order create and
+    # payment process. We are only interested in the "order.successful" event,
+    # which indicates successful payment.
+    # Later, we can switch to "hooks/webhook_url_on_success" - which will
+    # be triggered *only* by the "order.successful" event.
+    # (That webhook is not available at this time (October 18, 2017)).
+
+    payload = JSON.parse request.body.read
+
+    return head(:ok) unless payload['event'] == 'order.successful'
+
+    resource = HipsService.validate_webhook_origin(payload['jwt'])
+
+    payment_id = resource['merchant_reference']['order_id']
+    hips_id    = resource['id']
+
+    payment = Payment.find(payment_id)
+    payment.status = Payment.order_to_payment_status(resource['status'])
+    payment.save
+
+    log_hips_activity('Webhook', 'info', payment_id, hips_id)
+
+  rescue RuntimeError => exc
+    log_hips_activity('Webhook', 'error', payment_id, hips_id, exc)
```

Comment:
could replace `'Webhook'` with `__method__`

I'm not sure that's necessary, but just a thought.  To _me_ that makes the intent clearer.  But that might do the opposite for others.

Reply:
I'll keep it as is. It's faster this way and I don't have to worry about changing a symbol to a string.